common/persistence/visibility_triple_manager.go

```diff
@@ -373,14 +373,17 @@ func (v *visibilityTripleManager) getShadowMgrForDoubleRead(domain string) Visib
 	}
 
 	// Valid cases:
-	// case3: when it is double read, and both advanced visibility are available, and read mode is from destination
-	if v.readModeIsFromDestination(domain) {
-		return v.sourceVisibilityManager
-	}
-	// case4: when it is double read, and both advanced visibility are available, and read mode is from source
+	// case3: when it is double read, and both advanced visibility are available, and read mode is from source
+	// we first check readModeIsFromSource since we use this flag to control which is the primary visibility manager
 	if v.readModeIsFromSource(domain) {
```

Comment:
Why do we need those readModeIsFromSource and readModeIsFromDestinatio?

Reply:
Those are renamed from EnableReadVisibilityFromES/Pinot to make this more generic and can be reused for different migrations. For example in Pinot migration, it is migrating from ES to Pinot, ES will be the source and Pinot will be the destination.